realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1016,10 +1016,19 @@ public void executeTransaction(Transaction transaction) {
             transaction.execute(this);
             commitTransaction();
         } catch (RuntimeException e) {
```

Comment:
Ah, I misread it for Exception the first time. The reason we handled Error differently here was because we don't want people to catch and potentially ignore an OutOfMemoryError which can potentially corrupt a Realm file if you keep writing to it after that.

I think we added the RealmException to be able to document error scenarios better, i.e. you only need to catch RealmException, but I'm starting to wonder if that was a smart choice. If not, then just catching Throwable, cancel the transaction and rethrow should be fine.